src/Centreon/Infrastructure/Monitoring/Timeline/TimelineRepositoryRDB.php

```diff
@@ -0,0 +1,543 @@
+    ): string {
+        $request = $this->translateDbName("SELECT
+            a.acknowledgement_id AS `id`,
+            'acknowledgement' AS `type`,
+            a.entry_time AS `date`,
+            NULL AS `start_date`,
+            NULL AS `end_date`,
+            a.comment_data AS `content`,
+            c.contact_id AS `contact_id`,
+            a.author AS `contact_name`,
+            NULL AS `status_code`,
+            NULL AS `status_name`,
+            NULL AS `status_severity_code`,
+            NULL AS `tries`
+            FROM `:dbstg`.`acknowledgements` a
+            LEFT JOIN `:db`.contact AS `c` ON c.contact_alias = a.author
+            WHERE a.host_id = :host_id
+            AND (a.service_id = " . ($serviceId !== null ? ':service_id)' : '0 OR a.service_id IS NULL)') . "
+        ");
+
+        $collector->addValue(':host_id', $hostId, \PDO::PARAM_INT);
+        if ($serviceId !== null) {
+            $collector->addValue(':service_id', $serviceId, \PDO::PARAM_INT);
+        }
+
+        return $request;
+    }
+
+    /**
+     * get subquery to find acknowledgement events
+     *
+     * @param StatementCollector $collector
+     * @param integer $hostId
+     * @param integer|null $serviceId
+     * @return string subquery
+     */
+    private function prepareQueryForTimelineCommentEvents(
+        StatementCollector $collector,
+        int $hostId,
+        ?int $serviceId
+    ): string {
+        $request = $this->translateDbName("SELECT
+            c.comment_id AS `id`,
+            'comment' AS `type`,
+            c.entry_time AS `date`,
+            NULL AS `start_date`,
+            NULL AS `end_date`,
+            c.data AS `content`,
+            ct.contact_id AS `contact_id`,
+            c.author AS `contact_name`,
+            NULL AS `status_code`,
+            NULL AS `status_name`,
+            NULL AS `status_severity_code`,
+            NULL AS `tries`
+            FROM `:dbstg`.`comments` c
+            LEFT JOIN `:db`.contact AS `ct` ON ct.contact_alias = c.author
+            WHERE c.host_id = :host_id
+            AND (c.service_id = " . ($serviceId !== null ? ':service_id)' : '0 OR c.service_id IS NULL)') . "
+            AND source = 1
+        ");
+
+        $collector->addValue(':host_id', $hostId, \PDO::PARAM_INT);
+        if ($serviceId !== null) {
+            $collector->addValue(':service_id', $serviceId, \PDO::PARAM_INT);
+        }
+
+        return $request;
+    }
+
+    private function isAdmin(): bool
+    {
+        return ($this->contact !== null)
+            ? $this->contact->isAdmin()
+            : false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function setContact(ContactInterface $contact): TimelineRepositoryInterface
+    {
+        $this->contact = $contact;
+
+        return $this;
+    }
+
+    /**
+     * check if contact has enough rights to get events
+     *
+     * @return bool
+     */
+    private function hasNotEnoughRightsToContinue(): bool
```

Comment:
this method could be set into the TimelineService

Reply:
I let it cause this is needed to avoid sql errors, like empty `IN`
But, I rename it to `hasEnoughRightsToContinue` to avoid negation in method name (which is a bit confusing)